llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -23929,26 +24008,20 @@ bool RISCVTargetLowering::lowerDeinterleavedIntrinsicToVPLoad(
 
   const unsigned Factor = DeinterleaveResults.size();
 
-  auto *WideVTy = dyn_cast<ScalableVectorType>(Load->getType());
-  // TODO: Support fixed vectors.
-  if (!WideVTy)
+  auto *VTy = dyn_cast<VectorType>(DeinterleaveResults[0]->getType());
+  if (!VTy)
     return false;
 
-  unsigned WideNumElements = WideVTy->getElementCount().getKnownMinValue();
-  assert(WideNumElements % Factor == 0 &&
-         "ElementCount of a wide load must be divisible by interleave factor");
-  auto *VTy =
-      VectorType::get(WideVTy->getScalarType(), WideNumElements / Factor,
-                      WideVTy->isScalableTy());
```

Comment:
Does this assertion not still hold for fixed vectors?

Reply:
the loaded type could be wider than `VTy * Factor` so I think this assertion no longer holds. Plus, we're not deriving `VTy` from `WideVTy` like the deleted code shown here, but retrieving it from `DeinterleaveResults`, whose types have already been validated.